flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisStreamsSinkWriter.java

```diff
@@ -253,4 +263,26 @@ private boolean isRetryable(Throwable err) {
 
         return true;
     }
+
+    private String lookupStreamArn(final String streamName) {
+        if (streamArn.get() == null) {
+            try {
+                final String arn =
+                        kinesisClient
+                                .describeStream(
+                                        DescribeStreamRequest.builder()
+                                                .streamName(streamName)
+                                                .build())
+                                .get()
+                                .streamDescription()
+                                .streamARN();
```

Comment:
Is this possible backwards incompatible for users that are using the sink only today? After this change they might require additional IAM permissions?

Reply:
You're right. In this case, we can actually just extend the public API to allow users to specify streamARN instead